stack.hpp

```diff
@@ -16,19 +16,65 @@ class Stack {
     }
 
     void push(const T& element) {
-        if (position_ > capacity_) throw std::out_of_range("Not enough capacity");
-        storage_[position_++] = element;
+     /*   if (position_ < capacity_) // -1 add for the check *************
```

Comment:
There is no need to keep the old code in comments, if we ever want to go back, GIT has everything we need...